File: C#Assignment4/TaxiFare.cs
public class TaxiFare {
    public double getFare(double? distance)
    {
        if(distance==null||distance<=0){
            return -1;
        }

        double dist = (double) distance;
        
        if(distance<=2) return 50.00;
        if(distance<=5) return dist*20;
        if(distance<=15) return (5*20)+((dist-5)*18);
        return (5*20)+(10*18)+((dist-15)*15);  
        
    }
}

File: Assignment4/Assignment4 Python/TaxiFare.py
class TaxiFare:
    first_five_km = 5 * 20
    ten_km_after_first_five = 10 * 18

    # Fare Counter
    def get_fare(self, dist):
        if dist is None:
            return None;
        elif dist <= 0:
            return -1
        elif dist <= 2:
            return float(50)
        elif dist <= 5:
            return dist * 20
        elif dist <= 15:
            return self.first_five_km + (dist - 5) * 18
        return self.first_five_km + self.ten_km_after_first_five + (dist - 15) * 15